R/summarize_variables.R

```diff
@@ -478,90 +495,130 @@ s_summary.logical <- function(x,
+#'
+#' # comparison analysis - compare = TRUE
+#' a_summary(rnorm(10, 5, 1), .ref_group = rnorm(20, -5, 1), .var = "bla", compare = TRUE)
+#' a_summary(
+#'   factor(c("a", "a", "b", "c", "a")),
+#'   .ref_group = factor(c("a", "a", "b", "c")), compare = TRUE
+#' )
+#' a_summary(
+#'   c("A", "B", "A", "C"),
+#'   .ref_group = c("B", "A", "C"), .var = "x", compare = TRUE, verbose = FALSE
+#' )
+#' a_summary(
+#'   c(TRUE, FALSE, FALSE, TRUE, TRUE),
+#'   .ref_group = c(TRUE, FALSE), .in_ref_col = TRUE, compare = TRUE
+#' )
 #'
 #' @export
-a_summary.numeric <- make_afun(
-  s_summary.numeric,
-  .formats = .a_summary_numeric_formats,
-  .labels = .a_summary_numeric_labels
-)
+a_summary <- function(x,
+                      .N_col, # nolint
+                      .N_row, # nolint
+                      .var = NULL,
+                      .df_row = NULL,
+                      .ref_group = NULL,
+                      .in_ref_col = FALSE,
+                      .stats = names(get(paste0(".a_compare_", ifelse(is.numeric(x), "numeric", "counts"), "_labels"))),
+                      .formats = get(paste0(".a_compare_", ifelse(is.numeric(x), "numeric", "counts"), "_formats")),
+                      .labels = get(paste0(".a_compare_", ifelse(is.numeric(x), "numeric", "counts"), "_labels")),
+                      .indent_mods = get(paste0(".a_compare_", ifelse(is.numeric(x), "numeric", "counts"), "_indents")),
+                      na.rm = TRUE, # nolint
+                      na_level = NA_character_,
+                      compare = FALSE,
+                      ...) {
+  # Remove all-NA rows
+  if (!is.null(.df_row) && ncol(.df_row) > 1) {
+    in_tot_col <- nrow(.df_row) == length(x)
+    .df_row <- .df_row[rowSums(is.na(.df_row)) != ncol(.df_row), ]
+    if (in_tot_col && !identical(.df_row[[.var]], x)) x <- .df_row[[.var]]
+  }
 
-.a_summary_counts_formats <- summary_formats(type = "counts")
+  if (any(is.na(.df_row[[.var]])) && !any(is.na(x)) && !na.rm) levels(x) <- c(levels(x), "na-level")
+  if (!compare) {
+    x_stats <- s_summary(x = x, .N_col = .N_col, .N_row = .N_row, na.rm = na.rm, ...)
+    .stats <- setdiff(.stats, "pval")
+    .formats <- .formats[setdiff(names(.formats), "pval")]
+    .labels <- .labels[setdiff(names(.labels), "pval")]
+    if (!is.null(names(.indent_mods))) .indent_mods <- .indent_mods[setdiff(names(.indent_mods), "pval")]
+  } else {
+    x_stats <- s_compare(
+      x = x, .N_col = .N_col, .N_row = .N_row, na.rm = na.rm, .ref_group = .ref_group, .in_ref_col = .in_ref_col, ...
+    )
+  }
+  if (is.numeric(x)) {
+    for (i in intersect(.stats, c("mean_ci", "mean_pval", "median_ci", "quantiles"))) {
+      if (!i %in% names(.labels) || .labels[[i]] == .a_compare_numeric_labels[[i]]) {
+        .labels[[i]] <- attr(x_stats[[i]], "label")
+      }
+    }
+  }
+  if (length(.indent_mods) == 1 && is.null(names(.indent_mods))) {
+    .indent_mods <- rep(.indent_mods, length(.stats)) %>% `names<-`(.stats)
```

Comment:
```suggestion
    .indent_mods <- setNames(rep(.indent_mods, length(.stats)), .stats)
```
I would change this everywhere `names<-` is used as it is a bit harder to read imo